pyfr/plugins/solnwriter.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+from pyfr.h5writer import H5Writer
+from pyfr.inifile import Inifile
+from pyfr.plugins.base import BasePlugin
+
+
+class SolnWriterPlugin(BasePlugin):
+    name = 'solnwriter'
+    systems = ['*']
+
+    def __init__(self, intg, cfgsect, suffix=None):
+        super().__init__(intg, cfgsect, suffix)
+
+        basedir = self.cfg.getpath(cfgsect, 'basedir', '.')
+        basename = self.cfg.get(cfgsect, 'basename', raw=True)
+        serial_h5 = self.cfg.getbool(cfgsect, 'serial-h5', False)
```

Comment:
I suggest we move this to an environmental variable such as `PYFR_FORCE_SERIAL_HDF5` and move all of the logic into the writer class.  Otherwise every single plugin that wants to write something will need to copy this piece of boilerplate code and if you want to switch you need to set `serial-h5 = false` in your inifile for all plugins that write stuff.

As it is more of a debugging/performance thing it is probably best left to an environmental variable.